src/go/rpk/pkg/cli/cmd/topic/produce.go

```diff
@@ -123,6 +123,23 @@ func NewProduceCommand(fs afero.Fs) *cobra.Command {
 			defer cl.Close()
 			defer cl.Flush(context.Background())
 
+			var wg sync.WaitGroup
+			results := make(chan *kgo.Record, 55)
+
+			// Output via a channel & goroutine, to avoid concurrent fibers
+			// competing for stdout and mixing their lines together
+			go func() {
+				var outfBuf []byte
+				for {
+					r := <-results
```

Comment:
If a user runs this and decides to actually produce 0 records, this will hang indefinitely (here) and leak a goroutine. I don't think that's too critical here because the client is basically exiting immediately after.

Is it problematic to wrap the `AppendRecord` and `Stdout.Write` lines in a mutex? Under the hood, sending to a channel is basically using a mutex to append to a sized ringbuffer, so I wonder if it's easier to avoid the goroutine here and lock below:
```go
if outf != nil {
	mu.Lock()
	outfBuf = outf.AppendRecord(outfBuf[:0], r)
	os.Stdout.Write(outfBuf)
	mu.Unlock()
}
```

Reply:
Although I do like the change from returning on input read error. The current code (before this PR) doesn't actually wait for acknowledgement of all produced records; the WaitGroup addition here is a good idea.

Reply:
Promises are now guaranteed to be serial, so this PR is mostly unnecessary actually since I'm bumping franz-go. I'll keep the wait group aspect though.

Reply:
Actually, the wait group is also not necessary because there's a Flush existing. We can close this PR, the mangled output is addressed with an incoming franz-go bump.